containers/unit_tests/TestDynViewAPI_rank12345.hpp

```diff
@@ -20,4 +20,43 @@ namespace Test {
 TEST(TEST_CATEGORY, dyn_rank_view_api_operator_rank12345) {
   TestDynViewAPI<double, TEST_EXECSPACE>::run_operator_test_rank12345();
 }
+
+template <typename SharedMemorySpace>
+void test_dyn_rank_view_resize() {
+  int n = 1000000;
+  Kokkos::DynRankView<double, SharedMemorySpace> device_view("device view", n);
+  auto device_view_copy = device_view;
+
+  Kokkos::resize(device_view, 2 * n);
+
+  for (int i = 0; i < 2 * n; ++i) device_view(i) = i + 1;
+
+  Kokkos::fence();
+
+  for (int i = 0; i < 2 * n; ++i) ASSERT_EQ(device_view(i), i + 1);
```

Comment:
Please add a comment (in code) on the intent for the test.
Because out of context, one could expect your test should be doing other things, like asserting that dynrankview has the right size, that the data was preserved, that the data in excess is initialized, etc.